### PR TITLE
SorrTask332_Convert_Code_Review_gDoc_to_Markdown

### PR DESCRIPTION
**Pull request by zli00185**
This is the Markdown file after running linter. Would you mind taking a look and see which part need to be updated? @samarth9008

### CODE REVIEW

**Comment by samarth9008** [comment]
Go through this doc. https://github.com/sorrentum/sorrentum/blob/master/docs/First_review_process.md

Almost all the lines needed improvements.

See the doc and your PR to check if everything is consistent.

Refer to the linter file to generate TOC.

**Comment by samarth9008** [comment]
The best thing is to refer the code of other markdown files.

**Comment by zli00185** [comment]
@samarth9008 Could you check the submission?

**Comment by samarth9008** [comment]
Go through this doc. https://github.com/sorrentum/sorrentum/blob/master/docs/First_review_process.md

**Comment by zli00185** [comment]
Task #332

File: docs/Code_review.md [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+- Try to get a top to bottom review of a component once every N weeks (N = 2, 3)
+  - Sometimes the structure of the
+
+## Close the PR and delete the branch
+
+- When code is merged into master by one of the reviewers through the UI one can
+  select the delete branch option
+
+- Otherwise you can delete the branch using the procedure in 
+[[Git]{.underline}](https://docs.google.com/document/u/0/d/1zahC8uDnFGYRSgkBrQRgg3W3ZmDjZZJj6yln6YeuHq4/edit)
```
Comment (samarth9008) [inline docs/Code_review.md:278]: Improve the link formatting.

**Review by samarth9008 — COMMENTED** [review]
Make corrections and everything else looks good.

**Comment by samarth9008** [comment]
After corrections LGTM!

**Comment by samarth9008** [comment]
You have to hit the review button again. Read the doc carefully.

File: docs/Code_review.md [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+## Be clear in the PR request about what you want
+
+- Summarize what was done in the PR
+  - Refer to the GH task, but the task alone might not be sufficient
+  - A PR can implement only part of a complex task
+    - Which part is it implementing?
+    - Why is it doing it in a certain way?
+
+- If the code is not ready for merge, but you want a "pre-review" mark it as
+  "Draft PR"
```
Comment (DanilYachmenev) [inline docs/Code_review.md:113]: -> `...  but you want a "pre-review" convert PR to a draft`

**Comment by zli00185** [comment]
Sure. But the Reviewers section shows awaiting requested review from @DanilYachmenev and there isn't a re-request review button there for him.

File: docs/Code_review.md [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+## Workflow of a review in terms of GH labels
+
+- Please always add GP and Paul as reviewers, together with anybody else that
+  can be useful to review the code
+
+- The current meaning of the labels are:
+  - `PR_TO_REVIEW`: assigned by the automatic PR system / author
+  - `PR_CHANGE_REQUIRED`: assigned by Paul and GP if the PR needs some changes
+  - `PR_GP_APPROVED`: assigned by GP when he is ok with merging
+  - `PR_PAUL_APPROVED`: assigned by Paul when he is ok with merging
```
Comment (DanilYachmenev) [inline docs/Code_review.md:241]: replace this with a pointer to https://github.com/sorrentum/sorrentum/blob/master/docs/GitHub_ZenHub_workflows.md#pr-labels
like this text with a hyperlink
```
... are:
    - See GitHub ZenHub workflows doc
```

File: docs/Code_review.md [hunk 4]
```diff
@@ -0,0 +1,338 @@
...
+  - Other people that rely on your code are blocked
+
+## Ask for another review
+
+- Once you are done with resolving all the comments ask for another review
+
+## Workflow of a review in terms of GH labels
+
+- Please always add GP and Paul as reviewers, together with anybody else that
+  can be useful to review the code
```
Comment (DanilYachmenev) [inline docs/Code_review.md:235]: remove this one, obsolete
now we assign Team Leaders first and then the team leaders assign GP and Paul

File: docs/Code_review.md [hunk 5]
```diff
@@ -0,0 +1,338 @@
...
+
+- The current meaning of the labels are:
+  - `PR_TO_REVIEW`: assigned by the automatic PR system / author
+  - `PR_CHANGE_REQUIRED`: assigned by Paul and GP if the PR needs some changes
+  - `PR_GP_APPROVED`: assigned by GP when he is ok with merging
+  - `PR_PAUL_APPROVED`: assigned by Paul when he is ok with merging
+
+## Link PR to GH issue
+
+- TODO(Paul): Not sure what's the latest approach
```
Comment (DanilYachmenev) [inline docs/Code_review.md:245]: Replace with
```
Mention the corresponding issue in the PR description to ease the navigation
E.g., see an example
```
and add a hyperlink to https://github.com/sorrentum/sorrentum/pull/288#issue-1729654983

basically its a copy from https://github.com/sorrentum/sorrentum/blob/master/docs/First_review_process.md#mention-the-issue
which is not DRY but lets fix it later

**Review by DanilYachmenev — COMMENTED** [review]
almost there, gj!
see some nits and lgtm

File: docs/Code_review.md [hunk 6]
```diff
@@ -0,0 +1,338 @@
+<!-- toc -->
```
[the remark this answers is not in the document]
Comment (zli00185) [reply docs/Code_review.md:1]: Should I re-run the linter after adding it?

**Review by samarth9008 — APPROVED** [review]

**Comment by samarth9008** [comment]
LGTM

File: docs/Code_review.md [hunk 7]
```diff
@@ -0,0 +1,333 @@
...
+## Apply review comments everywhere
+
+- Apply a review comment everywhere, not just where the reviewer pointed out the
+  issue
+
+- E.g., if the reviewer says:
+  - Please replace:
+
+- `_LOG.warning("Hello %s".format(name))` with `_LOG.warning("Hello %s", name)`
+  you are expected to do this replacement:
```
Comment (DanilYachmenev) [inline docs/Code_review.md:191]: This does not look good when the document is viewed
probably it is a linter fix
lets do it like
```
- E.g., reviewer says:
  - "Please replace `_LOG.warning("Hello %s".format(name))` with `_LOG.warning("Hello %s", name)`"
- You are expected to do this replacement:
  - In the current review
  ...
```

**Review by DanilYachmenev — COMMENTED** [review]
1 fix, lgtm apart from that

File: docs/Code_review.md [hunk 8]
```diff
@@ -0,0 +1,331 @@
+# Code review
+<!-- toc -->
+
+- [General rules about code review](#general-rules-about-code-review)
+  * [Read the Google code review best practices](#read-the-google-code-review-best-practices)
+- [Code review workflows](#code-review-workflows)
+  * [Pull request](#pull-request)
+- [From the code author point of view\*\*](#from-the-code-author-point-of-view)
```
Comment (DanilYachmenev) [inline docs/Code_review.md:8]: remove `\*\*`

**Review by DanilYachmenev — COMMENTED** [review]
Well, 1 nit, 1 action and going to GP for merge

@zli00185 pls add a reference to this doc to `sorrentum/docs/README.md` file in `Project management` section right under `Documentation about guidelines` ref

it should look like this:
```
- Code review
    - PR review workflow from the code author and code reviewer points of view
```
where the first line is a hyperlink to this doc
see how it is done for other doc mentions in the readme

**Comment by gpsaggese** [comment]
Looks all done. Merging

**Pull request merged**